Sources/PenguinParallel/NonblockingThreadPool/NumberOperations.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Returns an array of all positive integers that are co-prime with `n`.
+///
+/// Two numbers are co-prime if their GCD is 1.
+internal func makeCoprimes(upTo n: Int) -> [Int] {
+  var coprimes = [Int]()
+  for i in 1...n {
+    var a = i
+    var b = n
+    // If GCD(a, b) == 1, then a and b are coprimes.
+    while b != 0 {
+      let tmp = a
+      a = b
+      b = tmp % b
+    }
+    if a == 1 { coprimes.append(i) }
+  }
+  return coprimes
+}
+
+/// Reduce `lhs` into `[0, size)`.
+///
+/// This is a faster variation than computing `x % size`. For additional context, please see:
+///     https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction
+internal func fastFit(_ lhs: Int, into size: Int) -> Int {
```

Comment:
Of course this should be an extension on `Int`, generic on `Integer`, and the parameter names should be improved… ;-)

Reply:
Definitely; will follow-up to pull this out. One question to ponder: what should the name of this be, and what operator should we define for it (if any)? ;-)

Reply:
Why would it have an operator rather than a long and descriptive name?  Surely this isn't used so often?
The answer to the name depends on your answer to my [question](https://github.com/saeta/penguin/pull/30/files/3f0ee97f56cd6add9990c25f54bccf07d1aad1a5#r429673124) about distribution.